### DISCUSSION TITLE
Sunbird-RC Frontend Setup Issue in Local

### DISCUSSION THREAD

**Discussion by shiva-rakshith** [Sunbird-RC/community#266] [Issues] [answered]
We are doing a complete(frontend and backend) sunbird-rc setup. The backend setup is done and APIs are running successfully.
we have tried to set up the sunbird-rc-ui by following the steps provided in the [documentation](https://docs.sunbirdrc.dev/use/setup-the-frontend) and made the configuration by referring to [edu core registry](https://github.com/ref-registries/edu-core-registries). 

we are facing the below error:
![image](https://user-images.githubusercontent.com/80666319/189891569-2788f4f8-40e7-4595-9f26-89c78affec06.png)

It seems the `config.json` file is missing. There is no `src/examples` folder as mentioned in the [documentation](https://docs.sunbirdrc.dev/use/setup-the-frontend) to refer for sample configuration. could you please guide me?

cc: @vrayulu @maheshkumargangula

**Comment by parthlawate** [comment]
@Pratikshakhandagale @code-blooded-guy @chaitrali-r can one of you help pls ?

**Comment by Pratikshakhandagale** [comment, marked as answer]
@shiva-rakshith Can you please follow [this](https://github.com/code-blooded-guy/sunbird-rc-ui/blob/main/README.md) setup the steps and let us know, if your facing any issue.

**Reply by Deeshantkotnala** [reply]
Hi,

While following the steps, there seems to be a problem in step 5.
In the project **(app.config.ts)**, inside switch **(line 44)** we don't have request unless we have config.json which should be generated by installation form but can't open the installation page. Even if we force open installation page, there is no function assigned to button click in html file. 
Can you confirm that this is problem or am I doing something wrong here.

Thanks

**Reply by Pratikshakhandagale** [reply]
@Deeshantkotnala have you tried to run project using step 7 and 8? Please try and share error with us which you are facing.

**Reply by Deeshantkotnala** [reply]
Yes, the page is blank, with error 
**ERROR TypeError: Cannot read properties of null (reading 'environment')
    at SafeSubscriber._next (app.config.ts:44:36)
    at SafeSubscriber.__tryOrUnsub (Subscriber.js:183:1)
    at SafeSubscriber.next (Subscriber.js:122:1)
    at Subscriber._next (Subscriber.js:72:1)
    at Subscriber.next (Subscriber.js:49:1)
    at MapSubscriber._next (map.js:35:1)
    at MapSubscriber.next (Subscriber.js:49:1)
    at FilterSubscriber._next (filter.js:33:1)
    at FilterSubscriber.next (Subscriber.js:49:1)
    at MergeMapSubscriber.notifyNext (mergeMap.js:70:1)**

The configuration in enironment.ts is
```
export const environment = {
  production: false,
  baseUrl: 'http://localhost:8081/api/',
  schemaUrl: 'assets/config/schema.json', //asset path OR URL
  logo: 'assets/images/logo.png', //asset path OR URL
  keycloakConfig: {
    url: 'http://localhost:8080/auth',
    realm: 'sunbird-rc',
    clientId: 'registry-frontend'
  }
};
```
